### PR TITLE
WSO2 Open Banking API Manager Puppet Scripts

### PR DESCRIPTION
**Pull request by KalanaDananjaya**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

   OS - Ubuntu 18.04
   Database - H2

### CODE REVIEW

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

**Review by Savidude — CHANGES_REQUESTED** [review]

**Review by Savidude — APPROVED** [review]

**Review by VimukthiPerera — APPROVED** [review]

File: modules/obam140/templates/carbon-home/repository/conf/datasources/master-datasources.xml.erb [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                     <url><%= @db_am_config_url %></url>
+                    <username><%= @db_am_config_username %></username>
+                    <password><%= @db_am_config_password %></password>
+                    <driverClassName><%= @db_am_config_driver %></driverClassName>
+                    <maxActive>150</maxActive>
+                    <maxWait>60000</maxWait>
+                    <testOnBorrow>true</testOnBorrow>
+                    <validationQuery><%= @db_am_config_validation_query %></validationQuery>
+                    <validationInterval>30000</validationInterval>
+                    <defaultAutoCommit>false</defaultAutoCommit>
```
[the remark this answers is not in the document]
Comment (KalanaDananjaya) [reply modules/obam140/templates/carbon-home/repository/conf/datasources/master-datasources.xml.erb:135]: Fixed

File: modules/obam140/templates/carbon-home/repository/conf/datasources/master-datasources.xml.erb [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        <datasource>
+            <name>WSO2AM_STATS_DB</name>
+            <description>The datasource used for getting statistics to API Manager</description>
+            <jndiConfig>
+                <name>jdbc/WSO2AM_STATS_DB</name>
+            </jndiConfig>
+            <definition type="RDBMS">
+                <configuration>
+		            <url><%= @db_apimgt_stat_url %></url>
```
[the remark this answers is not in the document]
Comment (KalanaDananjaya) [reply modules/obam140/templates/carbon-home/repository/conf/datasources/master-datasources.xml.erb]: Fixed

File: modules/obam140/manifests/params.pp [hunk 3]
```diff
@@ -0,0 +1,126 @@
+
```
[the remark this answers is not in the document]
Comment (KalanaDananjaya) [reply modules/obam140/manifests/params.pp]: Fixed

File: modules/obam140/manifests/init.pp [hunk 4]
```diff
@@ -0,0 +1,134 @@
+
```
[the remark this answers is not in the document]
Comment (KalanaDananjaya) [reply modules/obam140/manifests/init.pp]: Fixed

File: modules/obam140/templates/carbon-home/repository/deployment/server/executionplans/global_FreqPerDay.siddhiql.erb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+#----------------------------------------------------------------------------
+#  Copyright (c) 2019 WSO2, Inc. http://www.wso2.org
```
[the remark this answers is not in the document]
Comment (KalanaDananjaya) [reply modules/obam140/templates/carbon-home/repository/deployment/server/executionplans/global_FreqPerDay.siddhiql.erb]: Fixed

**Pull request merged**